org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/tracking/IssueTracker.java

```diff
@@ -55,7 +55,7 @@ public synchronized void matchAndTrackAsNew(String file, Collection<MutableTrack
    * @param trackables
    */
   public synchronized void matchAndTrackAsBase(String file, Collection<Trackable> trackables) {
-    Collection<MutableTrackable> current = cache.getCurrentTrackables(file);
+    Collection<Trackable> current = cache.getCurrentTrackables(file);
     if (current.isEmpty()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) NullPointerException might be thrown as 'current' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)